### PR TITLE
[Merged by Bors] - feat(Order/SuccPred/Limit): `IsSuccPrelimit.le_iff_forall_le`

### PR DESCRIPTION
**Pull request by vihdzp**
---

We also do some (hopefully uncontroversial) variable management on the side.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/SuccPred/Limit.lean [hunk 1]
```diff
@@ -289,12 +289,30 @@ end IsSuccArchimedean
...
+
+theorem IsSuccPrelimit.le_iff_forall_le (h : IsSuccPrelimit a) : a ≤ b ↔ ∀ c < a, c ≤ b := by
+  constructor
+  · intro ha c hc
+    exact hc.le.trans ha
+  · intro H
+    by_contra! ha
+    exact (lt_succ_of_not_isMax ha.not_isMax).not_le <| H _ (h.succ_lt ha)
+
+theorem IsSuccPrelimit.lt_iff_forall_lt (h : IsSuccPrelimit b) : a < b ↔ ∃ c < b, a < c := by
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Limit.lean:304]:
```suggestion
theorem IsSuccPrelimit.lt_iff_exists_lt (h : IsSuccPrelimit b) : a < b ↔ ∃ c < b, a < c := by
```

Comment (vihdzp) [reply Mathlib/Order/SuccPred/Limit.lean]: Nice catch!

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Order/SuccPred/Limit.lean [hunk 2]
```diff
@@ -289,12 +289,30 @@ end IsSuccArchimedean
 
 end PartialOrder
 
+section LinearOrder
+
+variable [LinearOrder α] [SuccOrder α]
+
+theorem IsSuccPrelimit.le_iff_forall_le (h : IsSuccPrelimit a) : a ≤ b ↔ ∀ c < a, c ≤ b := by
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Order/SuccPred/Limit.lean:296]: Otherwise LGTM. Thanks!
bors d+

[the remark this answers is not in the document]
Comment (vihdzp) [reply Mathlib/Order/SuccPred/Limit.lean:296]: Nice catch, this generalization works!

**Pull request closed without merging**